DLG System is ready for implementation

DrunkReaperMatt opened the pull request:
Plugin ready to merge

NB.
- **New** Blueprint Class from Rosetta Character
- **New** Game mode Blueprint 
- **New** Added Blueprint properties for dialogue trees files

DrunkReaperMatt committed:
DLG System is ready for implementation

karakune commented on Source/Rosetta/RosettaCharacter.cpp:
```diff
@@ -13,6 +13,10 @@
 
+#include "DlgContext.h"
+#include "DlgManager.h"
+#include "NonPlayerCharacter.h"
```
est-ce que ce include est nécessaire?

DrunkReaperMatt replied:
npc.h est superflu

removed

karakune commented on Source/Rosetta/RosettaCharacter.cpp:
```diff
@@ -194,6 +198,8 @@ void ARosettaCharacter::Interact()
 	{
 		InteractableHit->Interact();
+
+		//StartDialogue(Cast<ANonPlayerCharacter>(InteractableHit)->dialogueData, Cast<ANonPlayerCharacter>(InteractableHit));
```
À retirer

karakune commented on Source/Rosetta/RosettaCharacter.h:
```diff
@@ -101,4 +102,37 @@ class ARosettaCharacter : public ACharacter
+public:
+
+
+	// 	UPROPERTY(EditAnywhere, BlueprintReadOnly, Category = DlgData, meta = (AllowPrivateAccess = "true"))
```
À retirer?

DrunkReaperMatt replied:
est retirer

DrunkReaperMatt committed:
PATCH ajout plugin in git

karakune commented on Source/Rosetta/NonPlayerCharacter.h:
```diff
@@ -4,23 +4,42 @@
+	class ARosettaCharacter* player;
+
+public:
+	virtual void Interact()override;
```
add space after ) for clarity

karakune commented on Source/Rosetta/NonPlayerCharacter.cpp:
```diff
@@ -5,23 +5,42 @@
 
-void ANonPlayerCharacter::Interact()
-{
+/// Old Code from widget
```
delete if not-necessary

DrunkReaperMatt committed:
PATCH fix du empty folder plugin

karakune commented on Source/Rosetta/DialogueWidget.cpp:
```diff
@@ -90,50 +89,52 @@ void UDialogueWidget::GenerateWordWidgets(FString Sentence)
+	Player->SelectDialogueOption(DialogueStep);
+	UpdateOption(DialogueStep);
+
+	//	DialogueStep++;
```
delete if not-necessary

DrunkReaperMatt committed:
PULL REQUEST fixes ajouter selon review

DrunkReaperMatt committed:
FIX invalid function check

DrunkReaperMatt committed:
FIX invalid function check

DrunkReaperMatt committed:
Merge remote-tracking branch 'origin/feat-dialog-plugin' into feat-dialog-plugin

DrunkReaperMatt committed:
FIX repair missing references

DrunkReaperMatt committed:
ajout convo test

DrunkReaperMatt committed:
Start Integration Intro Quest

DrunkReaperMatt committed:
Begin integration de dialoque

DrunkReaperMatt committed:
Dialogue tree fonctonne

DrunkReaperMatt committed:
Finished first quest

- Add check pour completed quest
- Add temp dialogue pour quete completer
- Dialogue tree comporte correctement sur les reponses (bonne et les autre)

karakune commented on Source/Rosetta/DialogueWidget.cpp:
```diff
@@ -28,23 +28,23 @@ bool UDialogueWidget::Initialize()
 
-	UDialogueWidget::GenerateWordWidgets(DialogueIntroLine);
+	UpdateOption(DialogueStep);
+	Player->widget = this;
```
renommer widget pour quelque chose de plus explicite
